web/client/epics/wfsquery.js

```diff
@@ -154,7 +154,7 @@ export const wfsQueryEpic = (action$, store) =>
                 ...queryOptions
             };
             return Rx.Observable.merge(
-                getLayerJSONFeature({...layer, search: {...layer.search, url}}, ogcFilter, options)
+                getLayerJSONFeature({...layer, name: layer.name || action.filterObj.featureTypeName, search: {...layer.search, url}}, ogcFilter, options)
```

Comment:
Here I don't understand why the layer set by `setLayer` here doesn't work..., maybe you need to set layer before triggering click point Selector action? Give it a try. 
https://github.com/geosolutions-it/MapStore2/blob/e34489ca12b06e958f57052d068976bc7080059d/web/client/epics/identify.js#L355

Because I'd like to keep as much simpler as possible. If it doesn't work, please Can you add a unit test for this, specifying why you need this additional check.

Reply:
mmm... i tried #6363 with #6365 on this map.
http://localhost:8081/#/viewer/openlayers/10358

Tried to zoom to new york 
click here:
![image](https://user-images.githubusercontent.com/1279510/105012701-b7059280-5a3e-11eb-9b5b-db5c1c7efdb8.png)
Change to state layer in the selector and tried to edit. It doesn't work. I think you should try the suggested solution.

Reply:
i have tested the both pr together and the fix is to revert the change [here](https://github.com/geosolutions-it/MapStore2/pull/6412/files#diff-e8488af38db4e66cbfd306ca62e231eff32ea2138ca547086e739f19c91490beL74)

since i'm using only valid responses there

Reply:
I opened https://github.com/geosolutions-it/MapStore2/issues/6430 for fixing this structural issue.